src/journal-remote/journal-netlog-network.c

```diff
@@ -0,0 +1,201 @@
+static void format_rfc3339_timestamp(const struct timeval *tv, char *header_time, size_t header_size) {
+        char gm_buf[sizeof("+0530") + 1];
+        struct tm tm;
+        time_t t;
+
+        assert(header_time);
+
+        t = tv ? tv->tv_sec : ((time_t) (now(CLOCK_REALTIME) / USEC_PER_SEC));
+        localtime_r(&t, &tm);
+
+        strftime(header_time, header_size, "%Y-%m-%dT%T", &tm);
+
+        /* add fractional part */
+        if (tv)
+                snprintf(header_time + strlen(header_time), header_size, ".%06ld", tv->tv_usec);
+
+        /* format the timezone according to RFC */
+        xstrftime(gm_buf, "%z", &tm);
+        snprintf(header_time + strlen(header_time), header_size, "%.3s:%.2s ", gm_buf, gm_buf + 3);
+}
+
+/* The Syslog Protocol RFC5424 format :
+ * <pri>version sp timestamp sp hostname sp app-name sp procid sp msgid sp [sd-id]s sp msg
+ */
+int manager_push_to_network(Manager *m,
+                            int severity,
+                            int facility,
+                            const char *identifier,
+                            const char *message,
+                            const char *hostname,
+                            const char *pid,
+                            const struct timeval *tv) {
+        char header_priority[sizeof("< >1 ") + 1];
+        char header_time[FORMAT_TIMESTAMP_MAX];
+        uint16_t makepri;
+        struct iovec iov[13];
+        int n = 0;
+
+        assert(m);
+        assert(message);
+
+        makepri = (facility << 3) + severity;
+
+        /* First: priority field Second: Version  '<pri>version' */
+        snprintf(header_priority, sizeof(header_priority), "<%i>%i ", makepri, RFC_5424_PROTOCOL);
+        IOVEC_SET_STRING(iov[n++], header_priority);
+
+        /* Third: timestamp */
+        format_rfc3339_timestamp(tv, header_time, sizeof(header_time));
+        IOVEC_SET_STRING(iov[n++], header_time);
+
+        /* Fourth: hostname */
+        if (hostname)
+                IOVEC_SET_STRING(iov[n++], hostname);
+        else
+                IOVEC_SET_STRING(iov[n++], RFC_5424_NILVALUE);
+
+        IOVEC_SET_STRING(iov[n++], " ");
+
+        /* Fifth: identifier */
+        if (identifier)
+                IOVEC_SET_STRING(iov[n++], identifier);
+        else
+                IOVEC_SET_STRING(iov[n++], RFC_5424_NILVALUE);
+
+        IOVEC_SET_STRING(iov[n++], " ");
+
+        /* Sixth: procid */
+        if (pid)
+                IOVEC_SET_STRING(iov[n++], pid);
+        else
+                IOVEC_SET_STRING(iov[n++], RFC_5424_NILVALUE);
+
+        IOVEC_SET_STRING(iov[n++], " ");
+
+        /* Seventh: msgid */
+        IOVEC_SET_STRING(iov[n++], RFC_5424_NILVALUE);
+        IOVEC_SET_STRING(iov[n++], " ");
+
+        /* Eighth: [structured-data] */
+        IOVEC_SET_STRING(iov[n++], RFC_5424_NILVALUE);
+        IOVEC_SET_STRING(iov[n++], " ");
+
+        /* Ninth: message */
+        IOVEC_SET_STRING(iov[n++], message);
```

Comment:
we lose a '\0.*' here.
the same problem: https://github.com/systemd/systemd/pull/1460#commitcomment-13637787